http-filter-example/http_filter.cc

```diff
@@ -1,18 +1,60 @@
 #include <string>
+#include <unordered_map>
+#include <sstream>
+#include <vector>
 
 #include "http_filter.h"
 
+#include "source/common/common/logger.h"
 #include "envoy/server/filter_config.h"
 
 namespace Envoy {
 namespace Http {
 
 HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
     const sample::Decoder& proto_config)
-    : key_(proto_config.key()), val_(proto_config.val()) {}
+    : key_(proto_config.key()), val_(proto_config.val()), extra_(stoi(proto_config.extra())) {}
 
 HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
-    : config_(config) {}
+    : config_(config) {
+  // populate list of valid operations, implementation will be changed in the next few commits
+  operations_.insert("set-header");
```

Comment:
Nitpick: would be good to define magic strings as constants. Also the others below, e.g. `"no"`, `"op"`.

I was looking for the Envoy idiom, but I see a number of different ways they do it:

https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.h#L29C1-L41
https://github.com/envoyproxy/envoy/blob/main/source/common/http/http1/balsa_parser.cc#L29C1-L30C85
https://github.com/envoyproxy/envoy/blob/main/source/common/http/http2/codec_impl.h#L48

I'd avoid the last, it's going to be copied in every object file the header is included. I'd lean toward the first, since a directive like this is probably useful to multiple files/libraries. But since Envoy doesn't have a consistent idiom, anything is fine really.

The class is a bit strange to me, I'm not sure what the advantage is over a namespace and `extern const string_view`. But it's good to follow the project idioms 🤷